integration-tests/fetch-corpora.py

```diff
@@ -0,0 +1,168 @@
+#!/bin/env python3
+
+# not yet included
+#
+# XML file with references to files:
+# https://sturm-edition.de/api/files/
+# flat crawl:
+# https://celt.ucc.ie/publishd.html
+# deep crawlin':
+# https://gams.uni-graz.at/archive/objects/container:mws-gesamt/methods/sdef:Context/get?locale=fr&mode=&context=
+
+# Python 3.12 is required, but sys isn't imported yet
+
+import asyncio
+import io
+import os
+import tarfile
+from pathlib import Path
+from tempfile import TemporaryFile
+from typing import Final, NamedTuple
+
+from httpx import AsyncClient, HTTPError
+
+
+# constants
+
+
+class Archive(NamedTuple):
+    url: str
+    archive_documents_root: str
+    target_directory: str
+    extra_files: dict[str, str] = {}
+
+
+CORPORA_PATH: Final = Path(__file__).parent.resolve() / "corpora"
+
+
+ARCHIVE_DESCRIPTIONS: Final = (
```

Comment:
i was talking about populating `ARCHIVE_DESCRIPTIONS` from a multi-document yaml file, but that's just an idea, there is no real need for that.

Reply:
i see. but the problem is that this model only covers one form of provided data.

e.g. i also investigated the "Hopper Texts", and not only would the download usually abort, the tarball consisted of further tarballs. i ditched it also due to the actual contents. the point is that diverse sources will be hard to describe in one model.

Reply:
i usually solve this by beriving the model from the file name or something similar, but right now the integration code is so nice to look at, we should leave it as it is anyway without making it needlessly complicated.